src/strategies/l2-deversifi/index.ts

```diff
@@ -0,0 +1,49 @@
+import fetch from 'cross-fetch';
+import examplesFile from './examples.json';
+
+export const author = 'deversifi';
+export const version = '0.1.0';
+export const examples = examplesFile;
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const { api, token, limit = 300 } = options;
+
+  const pages = Math.ceil(addresses.length / limit);
+  const promises: any = [];
+
+  let api_url = api;
+  api_url += `?blockNumber=${snapshot}`;
+  api_url += `&token=${token}`;
+
+  Array.from(Array(pages)).forEach((x, i) => {
```

Comment:
Looks like here you are trying to make a API fetch call for each address, which won't scale for proposals with more addresses, is there any other approach you can try? Where you can get data in a single request

Reply:
This is not per address, but per 300 address (so if its more than 300 addresses requested it splits it into multiple requests).

Reply:
@plutoegg Cool then :) just noticed that, Thanks for letting me know